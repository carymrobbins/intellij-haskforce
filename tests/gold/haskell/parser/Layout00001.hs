module Layout00001 where

topLexer :: Lex a (Loc Token)
topLexer = do
                 t <- if bol' then lexBOL
                              else lexToken  -- t
                 el <- getSrcLocL
                 return 2
