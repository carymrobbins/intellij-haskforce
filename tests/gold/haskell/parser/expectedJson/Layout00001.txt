[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":20}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Layout00001"]},null,null]},[],[],[{"TypeSig":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":12,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[]},"topLexer"]}],{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":13},"srcInfoPoints":[]},"Lex"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},"srcInfoPoints":[]},"a"]}]}]},{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":29}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":20},"srcInfoPoints":[]},"Loc"]}]}]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":24},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":24},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":24},"srcInfoPoints":[]},"Token"]}]}]}]}]}]}]},{"PatBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[]},"topLexer"]}]},null,{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":9,"srcSpanStartColumn":10},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10}]},{"Do":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":4,"srcSpanEndLine":9,"srcSpanStartColumn":12},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":14,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":12},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1}]},[{"Generator":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":5,"srcSpanEndLine":6,"srcSpanStartColumn":18},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":20}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":18},"srcInfoPoints":[]},"t"]}]},{"If":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":5,"srcSpanEndLine":6,"srcSpanStartColumn":23},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":23},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":31},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":31}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":30,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":26},"srcInfoPoints":[]},"bol'"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":36},"srcInfoPoints":[]},"lexBOL"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":36},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":36},"srcInfoPoints":[]},"lexToken"]}]}]}]}]},{"Generator":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":21}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":18},"srcInfoPoints":[]},"el"]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":24},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":24},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":34,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":24},"srcInfoPoints":[]},"getSrcLocL"]}]}]}]},{"Qualifier":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":24,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":18},"srcInfoPoints":[]},"return"]}]}]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":25},"srcInfoPoints":[]},{"Int":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":8,"srcSpanEndLine":8,"srcSpanStartColumn":25},"srcInfoPoints":[]},2,"2"]}]}]}]}]]}]},null]}]]},[{"Comment":[false,{"srcSpanFilename":"A.hs","srcSpanEndColumn":50,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":46}," t"]}]]