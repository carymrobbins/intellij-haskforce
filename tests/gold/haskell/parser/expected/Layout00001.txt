Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Layout00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    GENDECL
      VARS
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('topLexer')
      PsiElement(HaskellTokenType.::)('::')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Lex')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
          ATYPE
            PsiElement(HaskellTokenType.()('(')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Loc')
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Token')
            PsiElement(HaskellTokenType.))(')')
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('topLexer')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          PsiElement(HaskellTokenType.do)('do')
          STMTS
            PAT
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('t')
            PsiElement(HaskellTokenType.<-)('<-')
            EXP
              PsiElement(HaskellTokenType.if)('if')
              EXP
                QVAR
                  QVARID
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('bol'')
              PsiElement(HaskellTokenType.then)('then')
              EXP
                QVAR
                  QVARID
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('lexBOL')
              PsiElement(HaskellTokenType.else)('else')
              EXP
                QVAR
                  QVARID
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('lexToken')
            PsiComment(HaskellTokenType.comment)('-- t\n')
            PAT
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('el')
            PsiElement(HaskellTokenType.<-)('<-')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('getSrcLocL')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('return')
              PsiElement(HaskellTokenType.integertoken)('2')